### PR TITLE
fix: better lookup for profile location, fixes #377

### LINKED ISSUE #377: [app] Public profile fetched from wrong url
My public profile is located at 

```
https://gaia.blockstack.org/hub/1Maw8BjWgj6MWrBCfupqQuWANthMhefb2v/0/profile.json
```
but `app` fetches it from 
```
https://gaia.blockstack.org/hub/1Maw8BjWgj6MWrBCfupqQuWANthMhefb2v/profile.json
```

Relevant code:
https://github.com/blockstack/ux/blob/master/packages/keychain/src/identity.ts#L111

### PR DESCRIPTION
**Pull request by hstove**
@friedger mind taking a look at this? You should be able to use the demo app to determine if the profile is fetched correctly. LMK if you have any questions, or don't have time to look at this. Thanks!

### CODE REVIEW

File: packages/keychain/src/utils/gaia.ts [hunk 1]
```diff
@@ -113,3 +116,27 @@ const makeGaiaAuthToken = ({ hubInfo, privateKey, gaiaHubUrl }: ConnectToGaiaOpt
...
+}: {
+  gaiaUrl: string;
+  identity: Identity;
+}) => {
+  const attempts = [
+    `${gaiaUrl}${identity.address}/profile.json`,
+    `${gaiaUrl}${identity.address}/0/profile.json`,
+    `${gaiaUrl}${identity.address}/1/profile.json`,
+  ];
+  const url = await promiseAny(attempts.map(u => returnURLIfOk(u)));
```
Comment (friedger) [inline packages/keychain/src/utils/gaia.ts]: Why don't you lookup the zonefile?

**Review by friedger — COMMENTED** [review]
As a temporary fix it would work.

**Comment by markmhendrickson** [comment]
@friedger Mind approving this PR if it looks good to you? We'll get it merged.

**Comment by friedger** [comment]
@markmhx Is there a plan to change the auth protocol? Or to support zonefiles in the future in connect?
What is the behaviour if the profile url can't be found?

**Comment by markmhendrickson** [comment]
Those are good questions for @hstove to answer given his approach here. 😄 

My impression from [your comment above](https://github.com/blockstack/ux/pull/460#pullrequestreview-439433303) was that this PR might be good as-is for merging (if only as a temporary fix). But we can keep it open if you'd prefer to address your questions before we merge.

**Review by friedger — APPROVED** [review]
LGTM and discuss the use of zonefiles separately

**Comment by markmhendrickson** [comment]
Great, assigning to @hstove for merge. He can follow up on your questions separately.

**Comment by markmhendrickson** [comment]
@hstove I can't seem to re-request review from @friedger for some reason. Mind giving it a shot?

**Review by friedger — APPROVED** [review]
Looks good!

**Pull request merged**